exporter/clickhouseexporter/config.go

```diff
@@ -85,19 +82,35 @@ func (cfg *Config) enforcedQueueSettings() exporterhelper.QueueSettings {
 	}
 }
 
-func (cfg *Config) buildDSN(database string) (string, error) {
-	dsn, err := url.Parse(cfg.Endpoint)
+func (cfg *Config) buildDBOptions(database string) (*clickhouse.Options, error) {
+	opts, err := clickhouse.ParseDSN(cfg.Endpoint)
 	if err != nil {
-		return "", errConfigInvalidEndpoint
+		return nil, errConfigInvalidDSN
+	}
+
+	if database != "" {
+		opts.Auth.Database = database
 	}
+
 	if cfg.Username != "" {
-		dsn.User = url.UserPassword(cfg.Username, cfg.Password)
+		opts.Auth.Username = cfg.Username
+		opts.Auth.Password = cfg.Password
 	}
-	dsn.Path = "/" + database
-	params := url.Values{}
-	for k, v := range cfg.ConnectionParams {
-		params.Set(k, v)
+
+	return opts, nil
+}
+
+func (cfg *Config) buildDB(database string) (*sql.DB, error) {
+	opts, err := cfg.buildDBOptions(database)
+	if err != nil {
+		return nil, err
 	}
-	dsn.RawQuery = params.Encode()
-	return dsn.String(), nil
+
+	conn := clickhouse.OpenDB(opts)
+	conn.SetMaxIdleConns(5)
```

Comment:
It is better to make connection pool as config settings.

Reply:
Done

Reply:
I think make it as config field, not just remove it. 

set `max_idle_conn` is useful when we want to use short connection for load balance.